river/tree/streaming_gradient_tree.py

```diff
@@ -0,0 +1,460 @@
+        self._n_splits = 0
+        self._n_node_updates = 0
+        self._n_observations = 0
+        self._depth = 0
+
+        self._objective: BaseObjective
+
+    def _target_transform(self, y):
+        """Apply transformation to the raw target input.
+
+        Different strategies are used for classification and regression. By default, implement
+        an identity function.
+        """
+        return y
+
+    def _update_features_stats(self, x: dict):
+        """ Update inner tree statistics with a new observation.
+
+        Keep estimators of the features' mean and variance.
+
+        Parameters
+        ----------
+        x
+            The incoming instance.
+        """
+
+        for feat_id, x_val in x.items():
+            # Skip nominal attributes
+            if (
+                not isinstance(x_val, numbers.Number)
+                or feat_id in self.nominal_attributes
+            ):
+                self.nominal_attributes.add(feat_id)
+                continue
+            self._features_mean_var[feat_id].update(x_val)
+
+    def _get_quantization_radius(self, feat_id: typing.Hashable):
+        """ Get the quantization radius for a given input feature. """
+        if self.quantization_strategy == self._CONSTANT_RAD:
+            return self.default_radius
+
+        if self._n_observations < 2:
+            # Default radius for quantization: might create too many bins at first
+            return self.default_radius
+
+        std_ = math.sqrt(self._features_mean_var[feat_id].get())
+
+        if math.isclose(std_, 0):
+            return self.default_radius
+
+        return std_ / self.quantization_radius_div
+
+    def _update_tree(self, x: dict, grad_hess: GradHess, w: float):
+        """ Update Streaming Gradient Tree with a single instance. """
+
+        leaf = self._root.sort_instance(x)
+        leaf.update(x, grad_hess, self, w)
+
+        if (
+            leaf.total_weight - leaf.last_split_attempt_at < self.grace_period
+            or leaf.depth >= self.max_depth
+        ):
+            return
+
+        # Update split attempt data
+        leaf.last_split_attempt_at = leaf.total_weight
+        best_split = leaf.find_best_split(self)
+
+        p = self._compute_p_value(best_split, leaf.total_weight)
+
+        if p < self.delta and best_split.loss_mean < 0:
+            leaf.apply_split(best_split, self)
+
+    @staticmethod
+    def _compute_p_value(split, n_observations):
+        # Null hypothesis: expected loss is zero
+        # Alternative hypothesis: expected loss is not zero
+
+        F = (
+            n_observations * (split.loss_mean * split.loss_mean) / split.loss_var
+            if split.loss_var > 0.0
+            else None
+        )
+
+        if F is None:
+            return 1.0
+
+        return 1 - FTest.cdf(F, 1, n_observations - 1)
+
+    def learn_one(self, x, y, *, w=1.0):
+        self._update_features_stats(x)
+
+        raw_pred = self._raw_prediction(x)
```

Comment:
Considering the signature of `compute_derivatives ` I suggest renaming these variables `y_true` and `y_pred` or something similar `y_raw_pred`, `y_true_trans`.

Reply:
Agreed! Changed in commit [940122d](https://github.com/online-ml/river/pull/458/commits/940122dfb81b6e62d7f43a2e51cfb113b90d899d).